node-api/handlers/beacon/validators.go

```diff
@@ -60,34 +53,24 @@ func (h *Handler) GetStateValidators(c handlers.Context) (any, error) {
 	}, nil
 }
 
-func (h *Handler) PostStateValidators(c handlers.Context) (any, error) {
-	req, err := utils.BindAndValidate[beacontypes.PostStateValidatorsRequest](
+func (h *Handler) GetStateValidators(c handlers.Context) (any, error) {
+	req, err := utils.BindAndValidate[beacontypes.GetStateValidatorsRequest](
 		c, h.Logger(),
 	)
 	if err != nil {
 		return nil, err
 	}
-	// TODO: remove this once status filter is implemented.
-	if len(req.Statuses) > 0 {
-		return nil, types.ErrNotImplemented
-	}
-	slot, err := utils.SlotFromStateID(req.StateID, h.backend)
-	if err != nil {
-		return nil, err
-	}
-	validators, err := h.backend.ValidatorsByIDs(
-		slot,
-		req.IDs,
-		req.Statuses,
+	return h.getStateValidators(req.StateID, req.IDs, req.Statuses)
+}
+
+func (h *Handler) PostStateValidators(c handlers.Context) (any, error) {
+	req, err := utils.BindAndValidate[beacontypes.PostStateValidatorsRequest](
+		c, h.Logger(),
 	)
 	if err != nil {
 		return nil, err
 	}
-	return beacontypes.ValidatorResponse{
-		ExecutionOptimistic: false, // stubbed
-		Finalized:           false, // stubbed
```

Comment:
can we introduce the logic for correctly returning these 2 fields as well? 
- I believed `Finalized` should always return true since the queried for slot (even "head") is guaranteed to be finalized by cometbft. Any raw slot # ahead of "head" should error out as its in the future anyways.
- Correspondingly `ExecutionOptimistic` I believe should always be false since we don't return data for any blocks that are not already Finalized.

Reply:
yes. Changed `Finalized` to true and `ExecutionOptimistic` was false earlier.